### DISCUSSION TITLE
FarmOS v1.0 - Farm automatically with a turtle!

### DISCUSSION THREAD

**Discussion by dabudke** [cc-tweaked/CC-Tweaked#616] [Show and tell]
I've been working on this for a while, and I'm proud to show it off!  This discussion will show you how to get started with FarmOS, and help you get started on your path to automated farming.

# Setup

Use these steps to set up your farming turtle with FarmOS, and get started with automated farming.

## Setting up your farm

Using the blocks put in slots 1-3, plan out your farm.  Till the ground, and plant seeds.  Then, set your turtle on the home block with, one block in front of it, a chest above and a chest below.

## Setting up your turtle

Setting up your turtle is easy, following the spreadsheet of slots.  All you need to do is put the items in the correct slots.

Spreadsheet: https://docs.google.com/spreadsheets/d/18fePAccYigIU1Mh0rI3OCEM_PavrcwJHAf9Q6mXUqzQ/edit?usp=sharing
Slot map:
![Slots go from 1 to 16, right to left and then top to bottom](https://user-images.githubusercontent.com/40306561/102932237-e4761500-446d-11eb-8853-547ac76a04d6.jpg)

To set up FarmOS, run this command: `pastebin run LmdhxBVW` and follow the prompts.

# Troubleshooting

## Home Obstructed

If you get the 'Home Obstructed!' message without a block in the way, do these steps:

1. Hold Control/Command + T until the output says `Terminated`.
2. Switch your fuel to the currently selected slot (it has a bigger outline)
3. Run the command `refuel 1`
4. Switch your coal back to the 4th slot
4. Reboot your turtle.

## Could not refuel!

If you keep hearing a low ding, this is why.  To fix this, all you need to do is put fuel in the fourth slot (you should be able to shift-click it in)

## Can't move (obstacle in the way?)

This issue is caused by a block in front, above and below the turtle.  I programed the turtle to be smart, so you shouldn't encounter this issue unless there is an issue with your farm setup.  Here is how you fix it:

### Turtle still on course:

1. Remove block from above/in front/below the turtle
2. Turtle should continue.

### Turtle not on course anymore

1. Terminate FarmOS by holding Control/Command + T
2. When presented with a prompt, run the commands `go <direction>` to return to your course
3. Reboot your turtle.

## Tutorial

If you want a step-by-step walkthrough on a blank world, go [here](https://imgur.com/a/NjD2PIR)

# Pastebin

Installer paste: `LmdhxBVW`
Program paste: `FbVzdCnM`

# Issues

To report bugs or suggest features or improvements, reply to this discussion.  Thank you for taking the time to read this through!

**Comment by kennethellixer123** [comment]
how do you get mods on minecraft eaglercraft javascrip runtime

**Reply by kennethellixer123** [reply]
pls help me
